src/cobra/core/reaction.py

```diff
@@ -416,15 +417,17 @@ def gene_reaction_rule(self):
         return self._gene_reaction_rule
 
     @gene_reaction_rule.setter
-    def gene_reaction_rule(self, new_rule):
+    def gene_reaction_rule(self, new_rule: str):
 
         # TODO: Do this :)
         if get_context(self):
             warn("Context management not implemented for " "gene reaction rules")
 
+        # This whole section just gets the gene names
         self._gene_reaction_rule = new_rule.strip()
```

Comment:
Are you talking about the comment?
It was removed in later commits, sorry about that.
If not, what are you talking about?

Reply:
Meant the `self._gene_reaction_rule = ...` part. Since `self` contains the rule this should be set with `self._gpr = GPR.from_string(...)`. Otherwise, the actual AST GPR won't be updated and you now have a case where `rxn._gpr` and `rxn._gene_reaction_rule` say different things. `rnx.gene_reaction_rule` just calls `str(self._gpr)` when used as a property. This enforces that the gpr and string rule are always in sync (single source of truth).

Reply:
Removed in the new code that uses the GPR()